flamby/benchmarks/benchmark_utils.py

```diff
@@ -198,22 +198,32 @@ def init_data_loaders(
     batch_size_test = batch_size if batch_size_test is None else batch_size_test
     if not pooled:
         training_dls = [
-            dl(
-                dataset(center=i, train=True, pooled=False),
-                batch_size=batch_size,
-                shuffle=True,
-                num_workers=num_workers,
-                collate_fn=collate_fn,
+            (
+                dl(
+                    center_dataset,
+                    batch_size=batch_size,
+                    shuffle=True,
+                    num_workers=num_workers,
+                    collate_fn=collate_fn,
+                )
+                if len(center_dataset := dataset(center=i, train=True, pooled=False))
+                > 0
+                else None
             )
             for i in range(num_clients)
         ]
         test_dls = [
-            dl(
-                dataset(center=i, train=False, pooled=False),
-                batch_size=batch_size_test,
-                shuffle=False,
-                num_workers=num_workers,
-                collate_fn=collate_fn,
+            (
+                dl(
+                    center_dataset,
+                    batch_size=batch_size_test,
+                    shuffle=False,
+                    num_workers=num_workers,
+                    collate_fn=collate_fn,
+                )
+                if len(center_dataset := dataset(center=i, train=False, pooled=False))
+                > 0
+                else None
```

Comment:
Is the code logic able to handle `None` afterwards ? such as in `fed_benchmark` ?

Reply:
It's fine if the answer is no

Reply:
currently that never happens